src/filter-context/category-context.js

```diff
@@ -0,0 +1,17 @@
+import { createContext, useContext, useState } from "react";
+
+
+const CategoryContext=createContext()
+
+
+const CategoryProvider=({children})=>{
+    const [category,setCategory]=useState([])
+    const [categoryName,setCategoryName]=useState([])
+    const [display,setDisplay]=useState("")
```

Comment:
you can take this into reducer as no of states are 3.

Reply:
yes will do it